batch/batch/front_end/query/query_v2.py

```diff
@@ -125,40 +125,46 @@ def parse_list_batches_query_v2(user: str, q: str, last_batch_id: Optional[int])
         where_conditions.append(f'({cond})')
         where_args += args
 
-    sql = f"""
-SELECT batches.*,
-  cancelled_t.cancelled IS NOT NULL AS cancelled,
+    sql = f"""\
+SELECT
+  batches.*,
+  cancelled_t.cancelled <=> 1 AS cancelled,
   job_groups_n_jobs_in_complete_states.n_completed,
   job_groups_n_jobs_in_complete_states.n_succeeded,
   job_groups_n_jobs_in_complete_states.n_failed,
   job_groups_n_jobs_in_complete_states.n_cancelled,
-  cost_t.cost, cost_t.cost_breakdown
-FROM job_groups
-LEFT JOIN batches ON batches.id = job_groups.batch_id
-LEFT JOIN billing_projects ON batches.billing_project = billing_projects.name
-LEFT JOIN job_groups_n_jobs_in_complete_states ON job_groups.batch_id = job_groups_n_jobs_in_complete_states.id AND job_groups.job_group_id = job_groups_n_jobs_in_complete_states.job_group_id
-LEFT JOIN LATERAL (
-  SELECT 1 AS cancelled
-  FROM job_group_self_and_ancestors
-  INNER JOIN job_groups_cancelled
-    ON job_group_self_and_ancestors.batch_id = job_groups_cancelled.id AND
-      job_group_self_and_ancestors.ancestor_id = job_groups_cancelled.job_group_id
-  WHERE job_groups.batch_id = job_group_self_and_ancestors.batch_id AND
-    job_groups.job_group_id = job_group_self_and_ancestors.job_group_id
-) AS cancelled_t ON TRUE
-STRAIGHT_JOIN billing_project_users ON batches.billing_project = billing_project_users.billing_project
-LEFT JOIN LATERAL (
-  SELECT COALESCE(SUM(`usage` * rate), 0) AS cost, JSON_OBJECTAGG(resources.resource, COALESCE(`usage` * rate, 0)) AS cost_breakdown
-  FROM (
-    SELECT resource_id, CAST(COALESCE(SUM(`usage`), 0) AS SIGNED) AS `usage`
+  cost_t.cost,
+  cost_t.cost_breakdown
+FROM batches
+INNER JOIN job_groups
+  ON batches.id = job_groups.batch_id
+INNER JOIN billing_projects
+  ON batches.billing_project = billing_projects.name
+INNER JOIN billing_project_users
+  ON batches.billing_project = billing_project_users.billing_project
```

Comment:
Also not in this PR, but thinking out loud - this took me a while to parse since it's not used elsewhere in the query and not in the return value. Eventually I noticed it's part of a where clause that's generated above and joined in below.

So thinking out loud, that feels a little icky because we have to remember to add this here _and_ we have to get the table names right for a where clause we happen to be adding below. I wonder whether we could change the `Query` interface from its current two parts (where clause, parameters) to three parts (join clause, where clause, parameters) to make that a little harder to mix up.

Reply:
Completely agree.